utils/ROI_Template.py

```diff
@@ -477,11 +558,50 @@ def append_to_container(self, container):
             info[self.namespace][MEASUREMENTS_KWD][self.toolType] = [clean_dict]
         else:
 
-            log.info(f"Appending to namespace {self.toolType}")
-            info[self.namespace][MEASUREMENTS_KWD][self.toolType].append(clean_dict)
+            if found_duplicate_roi(info[self.namespace][MEASUREMENTS_KWD][self.toolType]):
+                log.warning('Will not add duplicate')
+            else:
+                log.info(f"Appending to namespace {self.toolType}")
+                info[self.namespace][MEASUREMENTS_KWD][self.toolType].append(clean_dict)
 
         log.info("updating container...")
 
         container.update_info(info)
 
         pass
+
+
+    def found_duplicate_roi(self, existing_rois):
+
+        # We truncate to a length of 4, because sometimes an ROI will have a very long floating point coordinate
+        # Value, but when the user loads this into excel, it truncates it to like 6 or 8 decimal places, so we assume
+        # that 4 decimal places is small enough to be unique and still catch duplicates even with excell truncating.
+        n = 4
+        my_x1 = self.handle.start.x
+        my_y1 = self.handle.start.y
+        my_x2 = self.handle.end.x
+        my_y2 = self.handle.end.y
+
+        my_coords = [my_x1, my_y1, my_x2, my_y2]
+        my_coords = [truncate(c, n) for c in my_coords]
+
+        for roi in existing_rois:
+            roi_x1 = roi.get('handles', {}).get('start', {}).get('x', 0.0)
+            roi_y1 = roi.get('handles', {}).get('start', {}).get('y', 0.0)
+            roi_x2 = roi.get('handles', {}).get('end', {}).get('x', 0.0)
+            roi_y2 = roi.get('handles', {}).get('end', {}).get('y', 0.0)
+
+            roi_coords = [roi_x1, roi_x2, roi_y1, roi_y2]
+            roi_coords = [truncate(c, n) for c in roi_coords]
+
+            duplicated = [mc in roi_coords for mc in my_coords]
```

Comment:
Any way that the order of the coordinates could matter?  Like if there are these two ROIs, would they be duplicates?
```
ROI1 = {
    'start.x': 1.0,
    'start.y' : 2.0,
    'end.x': 3.0,
    'end.y': 4.0
}
ROI1 = {
    'start.x': 4.0,
    'start.y' : 2.0,
    'end.x': 1.0,
    'end.y': 3.0
}
```